### PR TITLE
Solve 1.4

### PR DESCRIPTION
**Pull request by qarux**

### CODE REVIEW

File: 1_concepts/1_4_cow/src/main.rs [hunk 1]
```diff
@@ -1,4 +1,28 @@
+use std::borrow::Cow;
+use std::env;
+
+const DEFAULT_PATH : &'static str = "/etc/app/app.conf";
+const VAR_CONFIG_KEY : &'static str = "APP_CONFIG";
+const CLI_CONFIG_ARG : &'static str = "--conf";
+
+fn get_path() -> Cow< 'static, str >
+{
+  if let Ok( value ) = env::var(VAR_CONFIG_KEY)
```
Comment (dmvict) [inline 1_concepts/1_4_cow/src/main.rs:10]: Please, update formatting.

File: 1_concepts/1_4_cow/src/main.rs [hunk 2]
```diff
@@ -1,4 +1,28 @@
+use std::borrow::Cow;
+use std::env;
+
+const DEFAULT_PATH : &'static str = "/etc/app/app.conf";
+const VAR_CONFIG_KEY : &'static str = "APP_CONFIG";
+const CLI_CONFIG_ARG : &'static str = "--conf";
```
Comment (dmvict) [inline 1_concepts/1_4_cow/src/main.rs:6]: Let's discuss the benefit of using CLI option and an environment variable as a const value.

File: 1_concepts/1_4_cow/src/main.rs (hunk 1 above)
Comment (qarux) [reply 1_concepts/1_4_cow/src/main.rs:10]: Done

Comment (qarux) [reply 1_concepts/1_4_cow/src/main.rs]: Done

File: 1_concepts/1_4_cow/src/main.rs (hunk 2 above)
Comment (qarux) [reply 1_concepts/1_4_cow/src/main.rs:6]: I can easily change it later and reuse it if needed.

Comment (dmvict) [reply 1_concepts/1_4_cow/src/main.rs:6]: I think that the code 
```rs
const DEFAULT_PATH : &'static str = "/etc/app/app.conf";
```
has sense. 

Code 
```rs
const VAR_CONFIG_KEY : &'static str = "APP_CONFIG";
``` 
also can be global. 

But CLI_CONFIG_ARG is more specific and should be in local function that handle specific case.

Comment (dmvict) [reply 1_concepts/1_4_cow/src/main.rs:6]: I see no mistakes. It's is only discuss about code.

**Comment by dmvict** [comment]
Can be merged.

**Pull request merged**